src/main/java/org/betonquest/betonquest/quest/event/time/TimeEventFactory.java

```diff
@@ -0,0 +1,95 @@
+package org.betonquest.betonquest.quest.event.time;
+
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.quest.event.Event;
+import org.betonquest.betonquest.api.quest.event.EventFactory;
+import org.betonquest.betonquest.api.quest.event.StaticEvent;
+import org.betonquest.betonquest.api.quest.event.StaticEventFactory;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.quest.event.DoNothingStaticEvent;
+import org.betonquest.betonquest.quest.event.NullStaticEventAdapter;
+import org.betonquest.betonquest.quest.event.PrimaryServerThreadEvent;
+import org.betonquest.betonquest.quest.event.worldselector.ConstantWorldSelector;
+import org.betonquest.betonquest.quest.event.worldselector.PlayerWorldSelector;
+import org.betonquest.betonquest.quest.event.worldselector.WorldSelector;
+import org.bukkit.Server;
+import org.bukkit.World;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.scheduler.BukkitScheduler;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Factory to create time events from {@link Instruction}s.
+ */
+public class TimeEventFactory implements EventFactory, StaticEventFactory {
+    /**
+     * Server to use for syncing to the primary server thread.
+     */
+    private final Server server;
+    /**
+     * Scheduler to use for syncing to the primary server thread.
+     */
+    private final BukkitScheduler scheduler;
+    /**
+     * Plugin to use for syncing to the primary server thread.
+     */
+    private final Plugin plugin;
+
+    /**
+     * Create the time event factory.
+     *
+     * @param server    server to use
+     * @param scheduler scheduler to use
+     * @param plugin    plugin to use
+     */
+    public TimeEventFactory(final Server server, final BukkitScheduler scheduler, final Plugin plugin) {
+        this.server = server;
+        this.scheduler = scheduler;
+        this.plugin = plugin;
+    }
+
+    @Override
+    public Event parseEvent(final Instruction instruction) throws InstructionParseException {
+        final String timeString = instruction.next();
+        final Time time = parseTimeType(timeString);
+        final long rawTime = parseTime(timeString);
```

Comment:
dont you need to cut off the first char? but at the same time i think it would not make any difference, if there would be no subtract (because subtract use `- - ` what results in `+`